liquid-staking/src/lib.rs

```diff
@@ -227,13 +225,59 @@ pub trait LiquidStaking<ContractReader>:
         }
     }
 
-    #[payable("*")]
     #[endpoint(unbondTokens)]
     fn unbond_tokens(&self) {
         self.blockchain().check_caller_is_user_account();
         let mut storage_cache = StorageCache::new(self);
         let caller = self.blockchain().get_caller();
+
+        require!(
+            self.is_state_active(storage_cache.contract_state),
+            ERROR_NOT_ACTIVE
+        );
+
+        let providers_to_unbond_from = self.unbond_from_provider(&caller);
+        let mut providers_unbond_from =
+            ManagedVec::<Self::Api, UnstakeTokenAttributes<Self::Api>>::new();
+        let current_epoch = self.blockchain().get_block_epoch();
+        let mut total_unstake_amount = BigUint::zero();
+
+        for unstake_token_attributes in providers_to_unbond_from.iter() {
+            if current_epoch < unstake_token_attributes.unbond_epoch {
+                continue;
+            }
+
+            let delegation_contract = unstake_token_attributes.delegation_contract.clone();
+            let unstake_amount = unstake_token_attributes.unstake_amount.clone();
+            let delegation_contract_mapper = self.delegation_contract_data(&delegation_contract);
+            let delegation_contract_data = delegation_contract_mapper.get();
+
+            if delegation_contract_data.total_unbonded_from_ls_contract < unstake_amount {
+                continue;
+            }
+
+            delegation_contract_mapper.update(|contract_data| {
+                contract_data.total_unstaked_from_ls_contract -= &unstake_amount;
+                contract_data.total_unbonded_from_ls_contract -= &unstake_amount
+            });
+
+            total_unstake_amount += unstake_amount;
+            providers_unbond_from.push(unstake_token_attributes);
```

Comment:
What's the use of this variable?

Reply:
not sure which one you refer to.. anyway the logic was changed